tests/tests/test_image_io.py

```diff
@@ -29,90 +29,53 @@ def image_array(request, array_2d, array_3d):
 
 
 @pytest.fixture()
-def shuffled_txt_path(tmp_path, array_3d):
+def txt_path(tmp_path, array_3d):
     """
     Return the path to a text file containing the paths of a series of 2D tiffs
-    in a random order
+    in order
     """
     txt_path = tmp_path / "imgs_file.txt"
-    write_tiff_sequence_with_txt_file(txt_path, array_3d)
-
-    # Shuffle paths in the text file into a random order
-    with open(txt_path, "r+") as f:
-        tiff_paths = f.read().splitlines()
-        random.Random(4).shuffle(tiff_paths)
-        f.seek(0)
-        f.writelines(line + "\n" for line in tiff_paths)
-        f.truncate()
-
-    return txt_path
-
-
-def write_tiff_sequence_with_txt_file(txt_path, image_array):
-    """
-    Write an image array to a series of tiffs, and write a text file
-    containing all the tiff file paths in order (one per line).
-
-    The tiff sequence will be saved to a sub-folder inside the same folder
-    as the text file.
-
-    Parameters
-    ----------
-    txt_path : pathlib.Path
-        Filepath of text file to create
-    image_array : np.ndarray
-        Image to write as sequence of tiffs
-    """
     directory = txt_path.parent
 
     # Write tiff sequence to sub-folder
     sub_dir = directory / "sub"
     sub_dir.mkdir()
-    save.to_tiffs(image_array, str(sub_dir / "image_array"))
+    save.to_tiffs(array_3d, str(sub_dir / "image"))
 
     # Write txt file containing all tiff file paths (one per line)
     tiff_paths = sorted(sub_dir.iterdir())
     txt_path.write_text(
         "\n".join([str(sub_dir / fname) for fname in tiff_paths])
     )
 
+    return txt_path
 
-def save_any(file_path, image_array):
-    """
-    Save image_array to given file path, using the save function matching
-    its file extension
 
-    Parameters
-    ----------
-    file_path : pathlib.Path
-        File path of image to save
-    image_array : np.ndarray
-        Image to save
+@pytest.fixture()
+def shuffled_txt_path(txt_path):
     """
-    if file_path.is_dir():
-        save.to_tiffs(image_array, str(file_path / "image_array"))
-
-    elif file_path.suffix == ".txt":
-        write_tiff_sequence_with_txt_file(file_path, image_array)
-
-    elif file_path.suffix == ".tif" or file_path.suffix == ".tiff":
-        save.to_tiff(image_array, str(file_path))
-
-    elif file_path.suffix == ".nrrd":
-        save.to_nrrd(image_array, str(file_path))
+    Return the path to a text file containing the paths of a series of 2D tiffs
+    in a random order
+    """
+    # Shuffle paths in the text file into a random order
+    with open(txt_path, "r+") as f:
+        tiff_paths = f.read().splitlines()
+        random.Random(4).shuffle(tiff_paths)
```

Comment:
@alessandrofelder this is actually the random seed, it just happened to be equal to the number of paths in this case! I set this to ensure we're getting the same result each time, and avoid the cases where it might shuffle the paths back into the correct order accidentally.

Reply:
Ha sorry! All good then, thanks!